text/0000-module-unification.md

```diff
@@ -0,0 +1,822 @@
+
+* `components` - component
+* `initializer` - initializer
+* `instance-initializers` - instance-initializer
+* `models` - model
+* `partials` - partial
+* `routes` - route
+* `services` - service
+* `transforms` - transform
+* `utils` - util
+
+The following private collections are allowed within collections (rule 5):
+
+* `components` - utils
+* `models` - utils
+* `initializers` - utils
+* `instance-initializers` - utils
+* `routes` - components, utils
+* `services` - utils
+* `transforms` - utils
+
+The following groups are proposed for collections (rule 6):
+
+* `data` - models, transforms
+* `init` - initializers, instance-initializers
+* `ui` - components, partials, routes
+
+The collection and type system is designed to be extensible, so that addons can
+contribute their own collections and types. The `data` collection and its
+corresponding types should be defined in ember-data. Liquid-fire might want to
+define an `animations` collection and a `transition` type, and expand `routes`
+to allow `animations` as a private collection.
+
+The specific format of collection and type declarations for addons is TBD.
+
+#### "Components" Collection
+
+This proposal broadens the scope of the term "component" to include all
+template-invocable parts of Ember. This includes today's components and helpers,
+and the future implementation of "glimmer components" (with angle brackets) and
+element modifiers.
+
+Grouping template-invocable elements together in a single collection recognizes
+that they already coexist in the same namespace. After all, only one helper OR
+component can be invoked as `{{foo-bar}}`. Using a common collection will not
+only simplify file management and searching, it will also provide implicit
+linting against creating a helper and class-based component of the same name.
+
+#### Private Collections
+
+You may wish to make a component available in a particular template without
+polluting the top-level `components` collection with a more local concern.
+Private collections allow you to augment a top-level collection's contents for
+use at a particular namespace.
+
+Private collections are declared as a directory sharing the name of the
+top-level collection, prefixed with a `-`. So the top-level `routes`
+collection could be augmented via a private `-components` collection.
+
+Say that you want to define a `post-viewer` component to be available only from
+within `src/ui/routes/posts/post/template.hbs`. You could achieve this by
+creating your component module in
+`src/ui/routes/posts/post/-components/post-viewer.js`.
+
+#### Non-resolved Files
+
+The rules above apply to modules that are resolved, namely `*.js` and `*.hbs`
+files. Other files that are used for documenting code, such as `*.md` and
+`*.html` files, can be freely co-located in any directories.
+
+Conventions will still be used for non-resolved files that have significance
+within an Ember project, including:
+
+* `src/ui/styles` - A project's stylesheets.
```

Comment:
Clarification.

Keeping styles separate from components preserves (4, relocatable within an application), but means moving components *between* applications (or refactoring a component into an engine and exporting) impossible without manually breaking-apart the style files.  Also seems to undermine 7 ("Ideally we will have a place to put tests and styles alongside corresponding components."). 

Is this really the intention, or are we supposed to do one of the following:
- Replicate the application structure in styles, e.g. `src/ui/styles/components/date-pocker/style.scss`?
- Optionally place styles along-side components (not shown in the examples), e.g. `src/ui/components/date-pocker/style.scss`?
Both of the above would allow me to refactor/move things around without manually breaking-down a monolithic `app.scss`.